src/main/scala/org/apache/mesos/chronos/scheduler/api/JobManagementResource.scala

```diff
@@ -173,32 +170,27 @@ class JobManagementResource @Inject()(val jobScheduler: JobScheduler,
           @PathParam("taskId") taskId: String,
           taskStat: TaskStat) : Response = {
     try {
-      var jobOpt = jobGraph.lookupVertex(jobName)
-      require(!jobOpt.isEmpty, "Job '%s' not found".format(jobName))
+      val jobOpt = jobGraph.lookupVertex(jobName)
+      require(jobOpt.nonEmpty, "Job '%s' not found".format(jobName))
       require(TaskUtils.isValidVersion(taskId), "Invalid task id format %s".format(taskId))
       require(jobOpt.get.dataProcessingJobType, "Job '%s' is not enabled to track data".format(jobName))
 
-      taskStat.numAdditionalElementsProcessed match {
-        case Some(num: Int) => {
+      taskStat.numAdditionalElementsProcessed.foreach {
+        num =>
           //NOTE: 0 is a valid value
           require(num >= 0,
             "numAdditionalElementsProcessed (%d) is not positive".format(num))
 
           jobStats.updateTaskProgress(jobOpt.get, taskId, num)
-        }
-        case None =>
       }
       Response.noContent().build
     } catch {
```

Comment:
We need curly braces here for multiple lines in the case class. Also, we want to make sure that there is no logical change here. See the scala style guide: http://docs.scala-lang.org/style/control-structures.html/ case - Omit braces if the case expression fits on a single line. Otherwise, use curly braces for clarity (even though they are not required by the parser).